utils/legacy_pointer/include/legacy_pointer.hpp

```diff
@@ -0,0 +1,259 @@
+
+    /**
+     * Creates a legacy_pointer_t from the given integer
+     * number
+     */
+    legacy_pointer_t(base_ptr_t u) : _contents(u){};
+  };
+
+  /* Whether if a pointer is null or not.
+   *
+   * A pointer is nullptr if the buffer id is 0,
+   * i.e the first BUFFER_ID_BITSIZE are zero
+   */
+  static inline bool is_nullptr(legacy_pointer_t ptr) {
+    return ((0x0000FFFFFFFFFFFFlu & ptr) == ptr);
+  }
+
+  /* Base nullptr
+   */
+  const legacy_pointer_t null_legacy_ptr = nullptr;
+
+  /* Data type to create buffer of byte-size elements
+   */
+  using buffer_data_type = uint8_t;
+
+  /* basic type for all buffers
+   */
+  using buffer_t = cl::sycl::buffer<buffer_data_type, 1>;
+
+  /* id of a buffer in the map
+   */
+  using buffer_id = short;
+
+  /* get_buffer_id
+   */
+  buffer_id get_buffer_id(legacy_pointer_t ptr) const {
+    auto theId =
+        (ptr & 0xFFFF000000000000lu) >> (ADDRESS_BITS - BUFFER_ID_BITSIZE);
+    return theId;
+  }
+
+  /*
+   * get_buffer_offset
+   */
+  off_t get_offset(legacy_pointer_t ptr) const {
+    auto theOffset = (ptr & 0x0000FFFFFFFFFFFFlu);
+    return theOffset;
+  }
+
+  /**
+   * Constructs the PointerMapper structure.
+   */
+  PointerMapper()
+      : __pointer_list{}, rng_(std::random_device()()), uni_(1, 256){};
+
+  /**
+   * PointerMapper cannot be copied or moved
+   */
+  PointerMapper(const PointerMapper &) = delete;
+
+  /**
+  *	empty the pointer list
+  */
+  void clear() { __pointer_list.clear(); }
+
+  /* generate_id
+   * Generates a unique id for a buffer.
+   */
+  buffer_id generate_id() {
+    // Limit the number of attemts to half the combinations
+    // just to avoid an infinite loop
+    int numberOfAttempts = 1ul << (BUFFER_ID_BITSIZE / 2);
+    buffer_id bId;
+    do {
+      bId = uni_(rng_);
+    } while (__pointer_list.find(bId) != __pointer_list.end() &&
+             numberOfAttempts--);
+    return bId;
+  }
+
+  /* add_pointer.
+   * Adds a pointer to the map and returns the fake pointer id.
+   * This will be the bufferId on the most significant bytes and 0 elsewhere.
+   */
+  legacy_pointer_t add_pointer(buffer_t &&b) {
+    auto nextNumber = __pointer_list.size();
+    buffer_id bId = generate_id();
+    __pointer_list.emplace(bId, b);
+    if (nextNumber > MAX_NUMBER_BUFFERS) {
+      return null_legacy_ptr;
+    }
+    base_ptr_t retVal = bId;
+    retVal <<= (ADDRESS_BITS - BUFFER_ID_BITSIZE);
+    return retVal;
+  }
+
+  /* get_buffer.
+   * Returns a buffer from the map using the buffer id
+   */
+  buffer_t get_buffer(buffer_id bId) const {
+    //  buffer_t retVal = __pointer_list.at(bId);
+    auto it = __pointer_list.find(bId);
+    if (it != __pointer_list.end()) {
+      return it->second;
+    }
+
+    std::cerr << "No sycl buffer has been found. Make sure that you have "
+                 "allocated memory for your buffer by calling malloc function."
+              << std::endl;
+    std::abort();
```

Comment:
Although I would normally agree with you, the project where this is used does not use exceptions, and they want things to just abort (not my decision!).